lib/ecto/changeset.ex

```diff
@@ -1349,6 +1349,43 @@ defmodule Ecto.Changeset do
     end
   end
 
+  import Ecto.Query # probably not the right place to do this?
+  # @spec  ??
+  def unsafe_validate_unique(changeset, field_names, repo, error_field \\ :base, error_message \\ "not unique") do
```

Comment:
API-wise, I wonder if it would be better to allow this:

    unsafe_validate_unique(changeset, :username, repo, "not unique")

If you want other considerations, then you can do this:

    unsafe_validate_unique(changeset, :username, &look_with_scope/1, "not unique")

where

     defp look_with_scope(query) do
       Repo.all from q in query, where: ...
     end

This way we don't need to support scoped queries... the user will rather do it directly. Passing a repo becomes just syntax sugar.